src/psyclone/tests/test_files/dynamo0p3/columnwise_op_app_w2v_kernel_mod.F90

```diff
@@ -0,0 +1,135 @@
+! DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+! FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+! DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+! SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+! CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+! OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+! OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+! -----------------------------------------------------------------------------
+!
+!> @brief Kernel which applies a columnwise assembled operator to a field
+!!        on W2V (discontinuous)
+
+module columnwise_op_app_w2v_kernel_mod
+
+use kernel_mod,              only : kernel_type
+use argument_mod,            only : arg_type, func_type,              &
+                                    GH_FIELD, GH_COLUMNWISE_OPERATOR, &
+                                    GH_READ, GH_INC,                  &
+                                    W2V, ANY_SPACE_2,                 &
+                                    GH_COLUMN_INDIRECTION_DOFMAP,     &
+                                    CELLS 
+
+use constants_mod,           only : r_def, i_def
+
+implicit none
+
+!-------------------------------------------------------------------------------
+! Public types
+!-------------------------------------------------------------------------------
+
+type, public, extends(kernel_type) :: columnwise_op_app_w2v_kernel_type
+  private
+  type(arg_type) :: meta_args(3) = (/                               &
+       arg_type(GH_FIELD,               GH_WRITE, W2V),             &  
+       arg_type(GH_FIELD,               GH_READ,  ANY_SPACE_2),     &
+       arg_type(GH_COLUMNWISE_OPERATOR, GH_READ,  W2V, ANY_SPACE_2) &
+       /)
+  integer :: iterates_over = CELLS
+contains
+  procedure, nopass :: columnwise_op_app_w2v_kernel_code
+end type columnwise_op_app_w2v_kernel_type
+
+!-------------------------------------------------------------------------------
+! Constructors
+!-------------------------------------------------------------------------------
+
+! Overload the default structure constructor for function space
+interface columnwise_op_app_w2v_kernel_type
+   module procedure columnwise_op_app_w2v_kernel_constructor
+end interface
+
+!-------------------------------------------------------------------------------
+! Contained functions/subroutines
+!-------------------------------------------------------------------------------
+public columnwise_op_app_w2v_kernel_code
+contains
+  
+  type(columnwise_op_app_w2v_kernel_type) function columnwise_op_app_w2v_kernel_constructor() result(self)
+    implicit none
+    return
+  end function columnwise_op_app_w2v_kernel_constructor
+
+  !> @brief The subroutine which is called directly from the PSY layer and
```

Comment:
I would remove the comments for brevity but I will not enforce it.

Reply:
I left them as I think they provide some useful information.